okhttp-dnsoverhttps/src/main/kotlin/okhttp3/dnsoverhttps/DnsOverHttps.kt

```diff
@@ -67,7 +67,11 @@ class DnsOverHttps internal constructor(
       }
     }
 
-    return lookupHttps(hostname)
+    return try {
+      listOf(InetAddressUtil.forString(hostname))
```

Comment:
Can you add a test?  

We need to avoid an exception in most calls here.  Maybe reverse this logic (try hostname first) and in the normal case call lookupHttps, and catch and exception, check if it's a ip address and then return forString.

Reply:
I think it would be better if we try converting by forString first because we can avoid a potential https connection if the hostname is an numerical IP address.

Reply:
If most of the calls will be hostname, maybe call the lookupHttps first would be better. I will update the code.